### PR TITLE
Add parameters to support change work queue

### PR DESCRIPTION
**Pull request by cwdsuzhou**
Fix virtual-kubelet/virtual-kubelet#840

### CODE REVIEW

**Comment by cwdsuzhou** [comment]
tests seem unhappy since https://github.com/virtual-kubelet/virtual-kubelet/pull/834 merged

File: node/podcontroller.go [hunk 1]
```diff
@@ -198,8 +217,9 @@ func NewPodController(cfg PodControllerConfig) (*PodController, error) {
 		ready:           make(chan struct{}),
 		done:            make(chan struct{}),
 		recorder:        cfg.EventRecorder,
-		k8sQ:            workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(), "syncPodsFromKubernetes"),
-		deletionQ:       workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(), "deletePodsFromKubernetes"),
+		k8sQ:            workqueue.NewNamedRateLimitingQueue(controllerRateLimiter(cfg.WorkQueueRetryQPS), "syncPodsFromKubernetes"),
```
Comment (sargun) [inline node/podcontroller.go:220]: Were the previous defaults 10?

**Comment by sargun** [comment]
tests should be fixed by https://github.com/virtual-kubelet/virtual-kubelet/pull/842 -- the problem is that K8s just release 1.17.7. This broke the validation script.

File: node/podcontroller.go (hunk 1 above)
Comment (cwdsuzhou) [reply node/podcontroller.go:220]: Yeap, it is defined 10 in the original client-go

**Comment by sargun** [comment]
btw, what do you recommend?

File: node/podcontroller.go (hunk 1 above)
Comment (cwdsuzhou) [reply node/podcontroller.go:220]: > btw, what do you recommend?

In our environment, we have changed it to 100 now to accelerate deleting pods.

**Comment by cwdsuzhou** [comment]
Before this works, we should wait for the PR https://github.com/virtual-kubelet/node-cli/pull/22 in node-cli to be merged

**Comment by sargun** [comment]
@cwdsuzhou I'll hold off on the merge. But, this to me looks like it should work, and fallback to 10.

**Comment by cpuguy83** [comment]
Currently we do not expose anything about the queue, or even that we use a queue to the package API. This changes that.

**Comment by cpuguy83** [comment]
I wonder if it would be better to expose the queue to the config instead of the queue config.
The downside here is we are stuck with the `workqueue.RatelimitingInterface` in that case... not that it's particularly bad, just need to be careful about what we expose.

**Comment by cwdsuzhou** [comment]
> @cwdsuzhou I'll hold off on the merge. But, this to me looks like it should work, and fallback to 10.

yes,  this should work

**Comment by cwdsuzhou** [comment]
> I wonder if it would be better to expose the queue to the config instead of the queue config.
> The downside here is we are stuck with the `workqueue.RatelimitingInterface` in that case... not that it's particularly bad, just need to be careful about what we expose.

This is better.

**Comment by cwdsuzhou** [comment]
@cpuguy83  @sargun 

PR updated, PTAL thanks

**Comment by cwdsuzhou** [comment]
> @cpuguy83 @sargun
> 
> PR updated, PTAL thanks

I will close https://github.com/virtual-kubelet/node-cli/pull/22 and re-open a new one when  this PR has been merged.

**Comment by sargun** [comment]
@cpuguy83 Are you okay with this, or do you want to take a different approach of queue / queue configuration / queue configuration policy?

**Pull request merged**